clang/test/SemaHLSL/Semantics/entry_parameter.hlsl

```diff
@@ -2,15 +2,18 @@
 // RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-mesh -hlsl-entry CSMain -x hlsl -finclude-default-header  -verify -o - %s
 
 [numthreads(8,8,1)]
-// expected-error@+3 {{attribute 'SV_GroupIndex' is unsupported in 'mesh' shaders, requires compute}}
-// expected-error@+2 {{attribute 'SV_DispatchThreadID' is unsupported in 'mesh' shaders, requires compute}}
-// expected-error@+1 {{attribute 'SV_GroupID' is unsupported in 'mesh' shaders, requires compute}}
-void CSMain(int GI : SV_GroupIndex, uint ID : SV_DispatchThreadID, uint GID : SV_GroupID) {
-// CHECK: FunctionDecl 0x{{[0-9a-fA-F]+}} <{{.*}}> line:[[@LINE-1]]:6 CSMain 'void (int, uint, uint)'
+// expected-error@+4 {{attribute 'SV_GroupIndex' is unsupported in 'mesh' shaders, requires compute}}
+// expected-error@+3 {{attribute 'SV_DispatchThreadID' is unsupported in 'mesh' shaders, requires compute}}
+// expected-error@+2 {{attribute 'SV_GroupID' is unsupported in 'mesh' shaders, requires compute}}
+// expected-error@+1 {{attribute 'SV_GroupThreadID' is unsupported in 'mesh' shaders, requires compute}}
+void CSMain(int GI : SV_GroupIndex, uint ID : SV_DispatchThreadID, uint GID : SV_GroupID, uint GThreadID : SV_GroupThreadID) {
```

Comment:
Why is this using mesh shader as the example of a stage where these are illegal?  These are actually legal for the mesh shader stage.

Reply:
@tex3d  Yes, you're right. I checked the spec https://microsoft.github.io/DirectX-Specs/d3d/MeshShader.html .  Those 4 attributes are legal for the mesh shader.

I think that they are legal for the amplification shader as well.

It looks like that we did something wrong in `SemaHLSL::CheckSemanticAnnotation` function
```
  switch (AnnotationAttr->getKind()) {
  case attr::HLSLSV_DispatchThreadID:
  case attr::HLSLSV_GroupIndex:
  case attr::HLSLSV_GroupThreadID:
  case attr::HLSLSV_GroupID:
    if (ST == llvm::Triple::Compute)
      return;
    DiagnoseAttrStageMismatch(AnnotationAttr, ST, {llvm::Triple::Compute});
    break;
```
We only check `Compute Shader` triple. I think that we should add checks for  `Mesh Shader` and `Amplification Shader` triples here. Right?

Reply:
We don't yet support those stages generally, and I don't think this will be the general diagnostic approach going forward anyway, so I'm not sure we need to update this code with those stages at this time.

Perhaps we can simply choose another shader stage in the test for which these are not allowed for now, such as `vertex`.

Reply:
Though you won't be able to use the same entry with the `[numthreads(...)]` attribute for a vertex shader.  Perhaps just add that as a define on the command line?  Or better yet, use a separate test file, since the shared content is extremely minimal between the tests.  That would certainly be less confusing.

Reply:
@tex3d It makes sense to me. I'll try to move the non-CS checks into `invalid_entry_parameter.hlsl`

Reply:
@tex3d Done. Updated the test in this commit https://github.com/llvm/llvm-project/pull/117781/commits/dc8d779f067e5b8d22e56036c4ba7320e297f339